ceph/ceph
# mgr/dashboard: allow users to preview OSDs before creating

Fixes: https://tracker.ceph.com/issues/42453
Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

## How to test
Unfortunately, this can't be tested with test_orchestrator. Currently, only cephadm support preview feature:
- Enable cephadm orchestrator backend
- Add some hosts. Some free disks are required.
- To test WAL/DB devices, disks must be in `SSD` type. To emulate this in VM, we can cheat the system:
  ```
  $ echo 0 > /sys/class/block/vda/queue/rotational
  ```
  Replace `vda` with correct device names.
- Run `ceph orch device ls --refresh` to update the data.
- Go to the OSD page and create some OSDs to see the preview.

## Screenshots

*OSD list*
![preview-list](https://user-images.githubusercontent.com/1691518/79865918-90aeba00-840e-11ea-9d03-59abcdf76a52.png)

*OSD detail*
![preview-details](https://user-images.githubusercontent.com/1691518/79865922-92787d80-840e-11ea-9fe5-6f0f21f4f988.png)



## Checklist

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>
